src/execute/oas3/parameter-builders.js

```diff
@@ -72,7 +76,7 @@ export function header({ req, parameter, value }) {
     return;
   }
 
-  if (parameter.content) {
+  if (value !== undefined && parameter.content) {
     const effectiveMediaType = Object.keys(parameter.content)[0];
 
     req.headers[parameter.name] = serialize(value, effectiveMediaType);
```

Comment:
We could also add something like this
```
if (value === undefined) {
   return value
}
```
to the `serialize` function
https://github.com/swagger-api/swagger-js/blob/b1e84804ee835eab7fe5671b9beefff61f871608/src/execute/oas3/content-serializer.js#L6

and then we would have to do
```
const serializedValue = serialize(value, effectiveMediaType)

if (serializedValue !== undefined) {
   req.headers[parameter.name] = serializedValue
}
```